### PR TITLE
CASMCMS-7993 Add procedure to customize iPXE binary names

### PR DESCRIPTION
**Pull request by ecozzi-hpe**
## Summary and Scope

_Summarize what has changed. Explain why this PR is necessary. What is impacted? Is this a new feature, critical bug fix, etc?_

Add procedure to customize iPXE binary names

## Issues and Related PRs

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves [CASMCMS-7993](https://jira-pro.its.hpecorp.net:8443/browse/CASMCMS-7993)

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [x] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: operations/boot_orchestration/Customize_iPXE_Binary_Names.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Customize iPXE Binary Names
+
+The default behavior for the cray-ipxe service is the build iPXE binaries with a well known name. However, to help
```
Comment (mharding-hpe) [inline operations/boot_orchestration/Customize_iPXE_Binary_Names.md:3]:
```suggestion
The default behavior for the cray-ipxe service is to build iPXE binaries with a well known name. However, to help
```

File: operations/boot_orchestration/Customize_iPXE_Binary_Names.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+prevent untrusted access to the iPXE binaries, sites may manually customize the iPXE binary names to a site specific
```
Comment (mharding-hpe) [inline operations/boot_orchestration/Customize_iPXE_Binary_Names.md:4]:
```suggestion
prevent untrusted access to the iPXE binaries, sites may manually customize the iPXE binary names to a site-specific
```

File: operations/boot_orchestration/Customize_iPXE_Binary_Names.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+## Procedure
+
+1. Edit the `cray-ipxe-settings` ConfigMap using one of the following options.
+
+   > **NOTE:** Save a backup of the ConfigMap before making any changes.
+
+   The following is an example of creating a backup:
+
+    ```bash
+    ncn-m001# kubectl get configmap -n services cray-ipxe-settings \
```
Comment (mharding-hpe) [inline operations/boot_orchestration/Customize_iPXE_Binary_Names.md:20]:
```suggestion
    ncn# kubectl get configmap -n services cray-ipxe-settings \
```

File: operations/boot_orchestration/Customize_iPXE_Binary_Names.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    -o yaml > /root/k8s/cray-ipxe-settings-backup.yaml
```
Comment (mharding-hpe) [inline operations/boot_orchestration/Customize_iPXE_Binary_Names.md:21]:
```suggestion
            -o yaml > /root/cray-ipxe-settings-backup.yaml
```

File: operations/boot_orchestration/Customize_iPXE_Binary_Names.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    ```
+
+   Within the configmap, edit the following keys to set the desired iPXE binary names.
+
+   | iPXE binary | Configmap Key Name | Default Value |
+   | ---| --- | --- |
+   | Regular iPXE | cray_ipxe_binary_name | ipxe.efi |
```
Comment (mharding-hpe) [inline operations/boot_orchestration/Customize_iPXE_Binary_Names.md:28]:
```suggestion
   | Regular iPXE | `cray_ipxe_binary_name` | `ipxe.efi` |
```

File: operations/boot_orchestration/Customize_iPXE_Binary_Names.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+   The following is an example of creating a backup:
+
+    ```bash
+    ncn-m001# kubectl get configmap -n services cray-ipxe-settings \
+    -o yaml > /root/k8s/cray-ipxe-settings-backup.yaml
+    ```
+
+   Within the configmap, edit the following keys to set the desired iPXE binary names.
+
+   | iPXE binary | Configmap Key Name | Default Value |
```
Comment (mharding-hpe) [inline operations/boot_orchestration/Customize_iPXE_Binary_Names.md:26]:
```suggestion
   | iPXE Binary | ConfigMap Key Name | Default Value |
```

File: operations/boot_orchestration/Customize_iPXE_Binary_Names.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+   | ---| --- | --- |
+   | Regular iPXE | cray_ipxe_binary_name | ipxe.efi |
+   | Debug iPXE | cray_ipxe_debug_binary_name | debug-ipxe.efi |
```
Comment (mharding-hpe) [inline operations/boot_orchestration/Customize_iPXE_Binary_Names.md:29]:
```suggestion
   | Debug iPXE | `cray_ipxe_debug_binary_name` | `debug-ipxe.efi` |
```

File: operations/boot_orchestration/Customize_iPXE_Binary_Names.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+   
+   > **NOTE:** Do not change the `cray_ipxe_binary_name_active` or
+   `cray_ipxe_debug_binary_name_active` keys in the
+   `cray-ipxe-settings` configmap. The cray-ipxe builder will automatically update these keys with the name of the currently built ipxe images once they are available.
```
Comment (mharding-hpe) [inline operations/boot_orchestration/Customize_iPXE_Binary_Names.md:33]:
```suggestion
   `cray-ipxe-settings` ConfigMap. The `cray-ipxe builder` will automatically update these keys with the name of the currently built iPXE images once they are available.
```

File: operations/boot_orchestration/Customize_iPXE_Binary_Names.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+    - **Option 1:** Edit the `cray-ipxe-settings` ConfigMap directly.
+
+      ```bash
+      ncn-m001#  kubectl edit configmap -n services cray-ipxe-settings
```
Comment (mharding-hpe) [inline operations/boot_orchestration/Customize_iPXE_Binary_Names.md:38]:
```suggestion
      ncn#  kubectl edit configmap -n services cray-ipxe-settings
```

File: operations/boot_orchestration/Customize_iPXE_Binary_Names.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+      ```
+
+    - **Option 2:** Edit the ConfigMap by saving the file, editing it, and reloading the ConfigMap.
+        1. Save the file.
+
+           ```bash
+           ncn-m001# kubectl get configmap -n services cray-ipxe-settings \
```
Comment (mharding-hpe) [inline operations/boot_orchestration/Customize_iPXE_Binary_Names.md:45]:
```suggestion
           ncn# kubectl get configmap -n services cray-ipxe-settings \
```

File: operations/boot_orchestration/Customize_iPXE_Binary_Names.md [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+           -o yaml > /root/k8s/cray-ipxe-settings.yaml
```
Comment (mharding-hpe) [inline operations/boot_orchestration/Customize_iPXE_Binary_Names.md:46]:
```suggestion
                   -o yaml > /root/cray-ipxe-settings.yaml
```

File: operations/boot_orchestration/Customize_iPXE_Binary_Names.md [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+           ```
+
+        2. Edit the cray-ipxe-settings.yaml file.
```
Comment (mharding-hpe) [inline operations/boot_orchestration/Customize_iPXE_Binary_Names.md:49]:
```suggestion
        2. Edit the `cray-ipxe-settings.yaml` file.
```

File: operations/boot_orchestration/Customize_iPXE_Binary_Names.md [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+
+           ```bash
+           ncn-m001# vi /root/k8s/cray-ipxe-bss-ipxe.yaml
```
Comment (mharding-hpe) [inline operations/boot_orchestration/Customize_iPXE_Binary_Names.md:52]:
```suggestion
           ncn# vi /root/cray-ipxe-bss-ipxe.yaml
```

File: operations/boot_orchestration/Customize_iPXE_Binary_Names.md [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+           ```
+
+        3. Reload the ConfigMap.
+
+           Deleting and recreating the ConfigMap will reload it.
+
+           ```bash
+           ncn-m001# kubectl delete configmap -n services cray-ipxe-settings
```
Comment (mharding-hpe) [inline operations/boot_orchestration/Customize_iPXE_Binary_Names.md:60]:
```suggestion
           ncn# kubectl delete configmap -n services cray-ipxe-settings
```

File: operations/boot_orchestration/Customize_iPXE_Binary_Names.md [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+           ncn-m001# kubectl create -f /root/k8s/cray-ipxe-settings.yaml
```
Comment (mharding-hpe) [inline operations/boot_orchestration/Customize_iPXE_Binary_Names.md:61]:
```suggestion
           ncn# kubectl create -f /root/cray-ipxe-settings.yaml
```

File: operations/boot_orchestration/Customize_iPXE_Binary_Names.md [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+           ```
+
+The cray-ipxe builder will detect the configuration change and rebuild the iPXE binaries within 30 to 90 seconds. Upon
```
Comment (mharding-hpe) [inline operations/boot_orchestration/Customize_iPXE_Binary_Names.md:64]:
```suggestion
The `cray-ipxe` builder will detect the configuration change and rebuild the iPXE binaries within 30 to 90 seconds. Upon
```

File: operations/boot_orchestration/Customize_iPXE_Binary_Names.md [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+successfully building the newly named binaries, the cray-ipxe builder will delete the old binaries from the shared files
```
Comment (mharding-hpe) [inline operations/boot_orchestration/Customize_iPXE_Binary_Names.md:65]:
```suggestion
successfully building the newly named binaries, the `cray-ipxe` builder will delete the old binaries from the shared file
```

**Review by mharding-hpe — CHANGES_REQUESTED** [review]

**Review by mharding-hpe — CHANGES_REQUESTED** [review]
Also, scroll to the bottom of the main PR page and see the markdown style and spelling linter checks. It scans any files being changed in the PR. Because Mikhail only recently added the linters, it will likely flag a lot of stuff in the files that wasn't being changed by your PR. However, the current goal is not to merge docs changes that run afoul of the linter, since otherwise we'd never get the old stuff cleaned up. 

Let me know if any of the linter complaints don't make sense to you and I can try to help out.

**Review by mharding-hpe — APPROVED** [review]
Looks good. Just need backport PRs to 1.2 and (if applicable) 1.0.

**Comment by mharding-hpe** [comment]
Eric, I'll try to help you out and make the backport PR for this

File: operations/index.md [hunk 18]
```diff
@@ -43,707 +44,774 @@ The following administrative topics can be found in this guide:
...
+- [Upgrade Compute Nodes with CRUS](compute_rolling_upgrades/Upgrade_Compute_Nodes_with_CRUS.md)
+- [Troubleshoot Nodes Failing to Upgrade in a CRUS Session](compute_rolling_upgrades/Troubleshoot_Nodes_Failing_to_Upgrade_in_a_CRUS_Session.md)
+- [Troubleshoot a Failed CRUS Session Because of Unmet Conditions](compute_rolling_upgrades/Troubleshoot_a_Failed_CRUS_Session_Due_to_Unmet_Conditions.md)
+- [Troubleshoot a Failed CRUS Session Because of Bad Parameters](compute_rolling_upgrades/Troubleshoot_a_Failed_CRUS_Session_Due_to_Bad_Parameters.md)
 
 <a name="configuration-management"></a>
 
 ## Configuration Management
 
+<<<<<<< HEAD
```
Comment (mharding-hpe) [inline operations/index.md:211]: This should not be in there

File: operations/index.md [hunk 19]
```diff
@@ -43,707 +44,774 @@ The following administrative topics can be found in this guide:
...
+  - [Update the Privacy Settings for Gitea Configuration Content Repositories](configuration_management/Update_the_Privacy_Settings_for_Gitea_Configuration_Content_Repositories.md)
+  - [Create and Populate a VCS Configuration Repository](configuration_management/Create_and_Populate_a_VCS_Configuration_Repository.md)
+- [Write Ansible Code for CFS](configuration_management/Write_Ansible_Code_for_CFS.md)
+  - [Target Ansible Tasks for Image Customization](configuration_management/Target_Ansible_Tasks_for_Image_Customization.md)
+- [CFS Key Management](configuration_management/CFS_Key_Management.md)
+
+The Configuration Framework Service \(CFS\) is available on systems for remote execution and configuration
+management of nodes and boot images.
+
+- [Configuration Management](configuration_management/Configuration_Management.md)
```
Comment (mharding-hpe) [inline operations/index.md:253]: This looks like duplicated content.

**Review by mharding-hpe — CHANGES_REQUESTED** [review]

**Comment by mharding-hpe** [comment]
I'm going to try to see if I can fix this up for you, Eric, since the problem is just with the index file, and I believe you just wanted to add an entry to that.,

File: operations/index.md (hunk 19 above)
Comment (ecozzi-hpe) [reply operations/index.md:214]: Done

**Comment by ecozzi-hpe** [comment]
The master branch has been updated for spelling and style changes that now conflict with the changes I have made in this PR. I'm going to close this PR and try again with a fresh PR.

**Pull request closed without merging**